src/device/ping-port-linux.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <ping-port.h>
+
+#include <cstdio>
+
+class PingPortLinux : public PingPort {
+public:
+  PingPortLinux(const char* filename);
+  ~PingPortLinux();
+
+  virtual int read(char* buffer, int nBytes) override final;
+  void sendBreak(int milliseconds);
+  bool setBaudrate(int baudrate);
+  virtual int write(const char* data, int nBytes) override final;
+
+private:
+  char _filename[256];
+  std::FILE* _handle;
+};
```

Comment:
There is libraries that handle serial ports for all OSs, avoid creating everything from the ground, this takes time and is not necessary.

Reply:
This is the path that took me the least time to test everything with the device class and ping360. We can do more implementations in the future.

Reply:
Its not the fastest way and I would hope to not improve such thing but remove completely. 
There is C++ std candidates for serial connections like [boost serial port](https://www.boost.org/doc/libs/1_65_0/doc/html/boost_asio/overview/serial_ports.html) or others projects around github.
We are unable to replicate the work, functionality and stability of such libraries with our team, creating libraries that already exist is unnecessary and time consuming.

Reply:
> Its not the fastest way and I would hope to not improve such thing but remove completely.

I never used boost. I never used qt serial port in a small application, I have this serial port application that works and well tested, I copy and pasted the code. That was fast. We can remove the linux-only implementation for boost in the future, but that will take time.